api/src/main/java/me/lokka30/treasury/api/economy/currency/Currency.java

```diff
@@ -1,236 +1,141 @@
-     * @param startingBalance       a function, giving a starting balance for the inputted player uuid, which can be null.
-     * @param balanceFormatter      a function, giving a human-readable format of a balance.
-     * @param names                 acceptable names of the currency. primary name shall be the first specified.
-     * @return new currency instance
-     * @author MrIvanPlays
+     * @return The currency's symbol.
+     * @author creatorfromhell
      * @since {@link me.lokka30.treasury.api.economy.misc.EconomyAPIVersion#v1_0 v1.0}
      */
-    @NotNull
-    public static Currency of(
-            @NotNull UUID currencyId,
-            @Nullable Character currencyChar,
-            int roundedDigits,
-            double conversionCoefficient,
-            @Nullable Function<UUID, Double> startingBalance,
-            @NotNull BiFunction<Double, Locale, String> balanceFormatter,
-            @NotNull String @NotNull ... names
-    ) {
-        Objects.requireNonNull(currencyId, "currencyId");
-        return new Currency(
-                currencyId,
-                currencyChar,
-                roundedDigits,
-                conversionCoefficient,
-                startingBalance,
-                balanceFormatter,
-                names
-        );
-    }
+    String getSymbol();
 
-    private final UUID currencyId;
-    private final Character currencyChar;
-    private final String[] names;
-    private final int roundedDigits;
-    private double conversionCoefficient;
-    private final Function<UUID, Double> startingBalance;
-    private final BiFunction<Double, Locale, String> balanceFormatter;
-
-    private Currency(
-            @Nullable UUID currencyId,
-            @Nullable Character currencyChar,
-            int roundedDigits,
-            double conversionCoefficient,
-            @Nullable Function<UUID, Double> startingBalance,
-            @NotNull BiFunction<Double, Locale, String> balanceFormatter,
-            @NotNull String @NotNull ... names
-    ) {
-        if (names.length < 1) {
-            throw new IllegalArgumentException("Empty array specified for currency name.");
-        }
-        this.balanceFormatter = Objects.requireNonNull(balanceFormatter, "balanceFormatter");
-        this.names = Objects.requireNonNull(names, "names");
-        this.currencyId = currencyId == null ? UUID.randomUUID() : currencyId;
-        this.currencyChar = currencyChar;
-        this.roundedDigits = roundedDigits;
-        this.conversionCoefficient = conversionCoefficient;
-        this.startingBalance = startingBalance == null ? ($) -> 0d : startingBalance;
-    }
+    /**
+     * Get's the currency's decimal character to be used for formatting purposes.
+     *
+     * @return The currency's decimal character.
+     * @author creatorfromhell
+     * @since {@link me.lokka30.treasury.api.economy.misc.EconomyAPIVersion#v1_0 v1.0}
+     */
+    char getDecimal();
```

Comment:
The decimal character isn't necesarily the same for everyone. It may be `.` or `,` depending on who the formatted string is shown to.

Reply:
> The decimal character isn't necesarily the same for everyone. It may be `.` or `,` depending on who the formatted string is shown to.

That's true - perhaps this method is unnecessary since the format method has a `Locale` argument anyways.